### DISCUSSION TITLE
Stargazers page: >5000 stars only shows as "5000+" stars

### DISCUSSION THREAD

**Discussion by laymonage** [community/community#63917] [Repositories]
### Select Topic Area

Bug

### Body

Not sure whether to post this in the Repository category or the Lists category, but I think the former fits better as even though it's related about stars, this isn't about the Lists feature.

On the main repository page, GitHub shows a good rounding of the number of stars, with the actual numbers shown when you hover on it:
<img width="375" alt="image" src="https://github.com/community/community/assets/6379424/3df55c6d-da8a-40ca-b4b3-13f8d0f98c47">

However, when viewing a repository's "Stargazers" page, e.g. https://github.com/giscus/giscus/stargazers, if the repository has more than 5000 stars, the star count only shows "5000+":

<img width="269" alt="image" src="https://github.com/community/community/assets/6379424/34ee8018-8232-4c0e-8fc0-77c19b1510fb">

This doesn't seem to make sense, or may be an oversight, as I would expect the Stargazers page to have more information on the stars. I don't see why there is a reason to truncate it to "5000+", it's not like counting stars is a hard/computationally expensive thing to do, given that you can show it on the main repository page.

Thus, I think it should show the actual number of stars.

Thanks!

**Comment by laymonage** [comment]
Bump.

**Comment by HariSekhon** [comment]
A good problem to have for sure with that popular a repo, I think one of mine might get there in another year or two the way it's going... which peaked my curiosity to click on this thread.

I agree that there is no good computer science reason I know of to cap the counter at 5000 when it doesn't cost more to just store and show the real number.

Looks like the real number is 7715 with a very smooth accumulation of stars, impressive!

[![Stargazers over time](https://starchart.cc/giscus/giscus.svg?variant=adaptive)](https://starchart.cc/giscus/giscus)

So the real number is tracked, it's just not shown due to UX decision by the look of it

**Reply by laymonage** [reply]
Thanks!

> I agree that there is no good computer science reason I know of to cap the counter at 5000 when it doesn't cost more to just store and show the real number.
> So the real number is tracked, it's just not shown due to UX decision by the look of it

Exactly. I mean, they even already show you the real number of stars on the main repo view when you hover on the number of stars on the star button:

<img width="224" alt="7,715 shown as a tooltip when you hover on the star button's number" src="https://github.com/user-attachments/assets/e4f4ef1f-e173-450d-969b-c23331ee5ade">


The main repo view is highly likely to be more computationally expensive to load than the stargazers view.

There's really no valid reason (if there was, not anymore) to truncate it to 5,000+ on the stargazers page.

**Comment by laymonage** [comment]
Bump.
